internal/api/lib_test.go

```diff
@@ -1055,3 +1061,100 @@ func TestCustomReflectQuerier(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "SMALL FRYS :)", response.Text)
 }
+
+// TestFloats is a port of the float_instrs_are_deterministic test in cosmwasm-vm
+func TestFloats(t *testing.T) {
+	type Value struct {
+		U32 *uint32 `json:"u32,omitempty"`
+		U64 *uint64 `json:"u64,omitempty"`
+		F32 *uint32 `json:"f32,omitempty"`
+		F64 *uint64 `json:"f64,omitempty"`
```

Comment:
Does this work well for zero values? E.g. how is 

```
val := Value {
    F64: 0
}
```

serialized?

Reply:
Yes, that's why I used pointers instead of direct integers. Missing ones will be `nil` and the given one a pointer to an int with value `0`.
In your example, this holds:
```go
v := new(uint64)
*v = 0
val := Value{
	F64: v,
}
j, _ := json.Marshal(val)
require.Equal(t, `{"f64":0}`, string(j))
```

Reply:
okay, great. Thanks for double checking